com.hazelcast.jet.kinesis.KinesisIntegrationTest [HZ-1185]

It was found, 28 tests fail in KinesisIntegrationTests, 14 in before, and 14 in cleanup phase.

https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-MacOS-ZuluJDK8/lastCompletedBuild/testReport/
 
Before:  
Error Message
com.hazelcast.jet.JetException: Abort waiting for stream listing, too many attempts
	at com.hazelcast.jet.kinesis.KinesisTestHelper.wait(KinesisTestHelper.java:219)
	at com.hazelcast.jet.kinesis.KinesisTestHelper.callSafely(KinesisTestHelper.java:213)
	at com.hazelcast.jet.kinesis.KinesisTestHelper.streamExists(KinesisTestHelper.java:76)
	at com.hazelcast.jet.kinesis.KinesisTestHelper.deleteStream(KinesisTestHelper.java:127)
	at com.hazelcast.jet.kinesis.AbstractKinesisTest.before(AbstractKinesisTest.java:85)

Clean up:
java.lang.NullPointerException
	at com.hazelcast.jet.core.JetTestSupport.cleanUpCluster(JetTestSupport.java:386)
	at com.hazelcast.jet.kinesis.AbstractKinesisTest.after(AbstractKinesisTest.java:93)

Example below
-----------------------------------------------------------------------

_master_ (commit 45d0f80a4771db41070503cef73475a083fc9caf )

Failed on MacOS-ZuluJDK8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-MacOS-ZuluJDK8/54/consoleFull

<details><summary>Stacktrace:</summary>

```
com.hazelcast.jet.JetException: Abort waiting for stream listing, too many attempts
	at com.hazelcast.jet.kinesis.KinesisTestHelper.wait(KinesisTestHelper.java:219)
	at com.hazelcast.jet.kinesis.KinesisTestHelper.callSafely(KinesisTestHelper.java:213)
	at com.hazelcast.jet.kinesis.KinesisTestHelper.streamExists(KinesisTestHelper.java:76)
	at com.hazelcast.jet.kinesis.KinesisTestHelper.deleteStream(KinesisTestHelper.java:127)
	at com.hazelcast.jet.kinesis.AbstractKinesisTest.before(AbstractKinesisTest.java:85)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:365)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:63)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:39)
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

</details>

<details><summary>Standard output:</summary>

```
Finished Running Test: initialRead_timestamp in 0.000 seconds.
Started Running Test: restart_staticStream_graceful
2022-04-01 20:30:16,078 [ WARN] [Time-limited test] [c.h.j.k.KinesisIntegrationTest]: Amazon SDK failure, ignoring and retrying.
com.amazonaws.SdkClientException: Unable to execute HTTP request: Connect to localhost:55571 [localhost/127.0.0.1] failed: Connection refused (Connection refused)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleRetryableException(AmazonHttpClient.java:1216) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1162) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:811) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:779) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:753) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:713) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:695) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:559) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:539) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.services.kinesis.AmazonKinesisClient.doInvoke(AmazonKinesisClient.java:2980) ~[aws-java-sdk-kinesis-1.12.150.jar:?]
	at com.amazonaws.services.kinesis.AmazonKinesisClient.invoke(AmazonKinesisClient.java:2947) ~[aws-java-sdk-kinesis-1.12.150.jar:?]
	at com.amazonaws.services.kinesis.AmazonKinesisClient.invoke(AmazonKinesisClient.java:2936) ~[aws-java-sdk-kinesis-1.12.150.jar:?]
	at com.amazonaws.services.kinesis.AmazonKinesisClient.executeListStreams(AmazonKinesisClient.java:1752) ~[aws-java-sdk-kinesis-1.12.150.jar:?]
	at com.amazonaws.services.kinesis.AmazonKinesisClient.listStreams(AmazonKinesisClient.java:1721) ~[aws-java-sdk-kinesis-1.12.150.jar:?]
	at com.amazonaws.services.kinesis.AmazonKinesisClient.listStreams(AmazonKinesisClient.java:1764) ~[aws-java-sdk-kinesis-1.12.150.jar:?]
	at com.hazelcast.jet.kinesis.KinesisTestHelper.listStreams(KinesisTestHelper.java:158) ~[test-classes/:?]
	at com.hazelcast.jet.kinesis.KinesisTestHelper.callSafely(KinesisTestHelper.java:188) [test-classes/:?]
	at com.hazelcast.jet.kinesis.KinesisTestHelper.streamExists(KinesisTestHelper.java:76) [test-classes/:?]
	at com.hazelcast.jet.kinesis.KinesisTestHelper.deleteStream(KinesisTestHelper.java:127) [test-classes/:?]
	at com.hazelcast.jet.kinesis.AbstractKinesisTest.before(AbstractKinesisTest.java:85) [test-classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_302]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_302]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_302]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_302]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59) [junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit-4.13.2.jar:4.13.2]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56) [junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33) [junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24) [junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:365) [hazelcast-5.2-SNAPSHOT-tests.jar:5.2-SNAPSHOT]
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37) [hazelcast-5.2-SNAPSHOT-tests.jar:5.2-SNAPSHOT]
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104) [hazelcast-5.2-SNAPSHOT-tests.jar:5.2-SNAPSHOT]
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:63) [hazelcast-5.2-SNAPSHOT-tests.jar:5.2-SNAPSHOT]
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306) [junit-4.13.2.jar:4.13.2]
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100) [junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366) [junit-4.13.2.jar:4.13.2]
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103) [junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50) [hazelcast-5.2-SNAPSHOT-tests.jar:5.2-SNAPSHOT]
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29) [hazelcast-5.2-SNAPSHOT-tests.jar:5.2-SNAPSHOT]
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331) [junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79) [junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329) [junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66) [junit-4.13.2.jar:4.13.2]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293) [junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit-4.13.2.jar:4.13.2]
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:39) [hazelcast-5.2-SNAPSHOT-tests.jar:5.2-SNAPSHOT]
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66) [hazelcast-5.2-SNAPSHOT-tests.jar:5.2-SNAPSHOT]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299) [junit-4.13.2.jar:4.13.2]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293) [junit-4.13.2.jar:4.13.2]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_302]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_302]
Caused by: org.apache.http.conn.HttpHostConnectException: Connect to localhost:55571 [localhost/127.0.0.1] failed: Connection refused (Connection refused)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:156) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376) ~[httpclient-4.5.13.jar:4.5.13]
	at sun.reflect.GeneratedMethodAccessor105.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_302]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_302]
	at com.amazonaws.http.conn.ClientConnectionManagerFactory$Handler.invoke(ClientConnectionManagerFactory.java:76) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.conn.$Proxy43.connect(Unknown Source) ~[?:?]
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56) ~[httpclient-4.5.13.jar:4.5.13]
	at com.amazonaws.http.apache.client.impl.SdkHttpClient.execute(SdkHttpClient.java:72) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1343) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1154) ~[aws-java-sdk-core-1.12.150.jar:?]
	... 50 more
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.net.PlainSocketImpl.socketConnect(Native Method) ~[?:1.8.0_302]
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350) ~[?:1.8.0_302]
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206) ~[?:1.8.0_302]
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188) ~[?:1.8.0_302]
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[?:1.8.0_302]
	at java.net.Socket.connect(Socket.java:607) ~[?:1.8.0_302]
	at org.apache.http.conn.socket.PlainConnectionSocketFactory.connectSocket(PlainConnectionSocketFactory.java:75) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376) ~[httpclient-4.5.13.jar:4.5.13]
	at sun.reflect.GeneratedMethodAccessor105.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_302]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_302]
	at com.amazonaws.http.conn.ClientConnectionManagerFactory$Handler.invoke(ClientConnectionManagerFactory.java:76) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.conn.$Proxy43.connect(Unknown Source) ~[?:?]
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83) ~[httpclient-4.5.13.jar:4.5.13]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56) ~[httpclient-4.5.13.jar:4.5.13]
	at com.amazonaws.http.apache.client.impl.SdkHttpClient.execute(SdkHttpClient.java:72) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1343) ~[aws-java-sdk-core-1.12.150.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1154) ~[aws-java-sdk-core-1.12.150.jar:?]
	... 50 more
2022-04-01 20:30:16,080 [ INFO] [Time-limited test] [c.h.j.k.KinesisIntegrationTest]: Waiting for stream listing ...
2022-04-01 20:30:16,429 [ WARN] [Time-limited test] [c.h.j.k.KinesisIntegrationTest]: Amazon SDK failure, ignoring and retrying.
com.amazonaws.SdkClientException: Unable to execute HTTP request: Connect to localhost:55571 [localhost/127.0.0.1] failed: Connection refused (Connection refused)
...
```

</details>

--- a/extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/AbstractKinesisTest.java
+++ b/extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/AbstractKinesisTest.java
@@ -90,12 +90,16 @@ public abstract class AbstractKinesisTest extends JetTestSupport {
 
     @After
     public void after() {
-        cleanUpCluster(cluster);
+        if (cluster != null) {
+            cleanUpCluster(cluster);
+        }
 
         helper.deleteStream();
 
-        results.clear();
-        results.destroy();
+        if (results != null) {
+            results.clear();
+            results.destroy();
+        }
     }
 
     protected HazelcastInstance hz() {

